src/Language/PureScript/TypeChecker.hs

```diff
@@ -383,6 +388,11 @@ typeCheckModule (Module ss coms mn decls (Just exps)) = warnAndRethrow (addHint
     exports (PositionedDeclarationRef _ _ r1) r2 = exports r1 r2
     exports r1 (PositionedDeclarationRef _ _ r2) = exports r1 r2
     exports _ _ = False
+    -- We avoid Eq for `nub`bing as the dctor part of `TypeRef` evaluates to
+    -- `error` for the values generated here (we don't need them anyway)
+    nubEq (TypeRef pn1 _) (TypeRef pn2 _) = pn1 == pn2
+    nubEq r1 r2 = r1 == r2
+
```

Comment:
The addition of `nub`bing here isn't specifically related to the other changes, just a little improvement to prevent a type from being printed multiple times in the error message in some situations.